cabal-install/src/Distribution/Client/Main.hs

```diff
@@ -1320,3 +1327,33 @@ manpageAction commands flags extraArgs _ = do
           then dropExtension pname
           else pname
   manpageCmd cabalCmd commands flags
+
+pathAction :: PathFlags -> [String] -> Action
+pathAction pathflags _extraArgs globalFlags = do
+  let verbosity = fromFlag (pathVerbosity pathflags)
+  cfg <- loadConfig verbosity mempty
+  let getDir getDefault getGlobal =
+        maybe
+          getDefault
+          pure
+          (flagToMaybe $ getGlobal $ savedGlobalFlags cfg)
+      dirs =
+        [ ("cache-dir", getDir defaultCacheDir globalCacheDir)
+        , ("logs-dir", getDir defaultLogsDir globalLogsDir)
+        , ("store-dir", getDir defaultStoreDir globalStoreDir)
+        , ("config-file", getConfigFilePath (globalConfigFile globalFlags))
```

Comment:
I notice this list of open spellings (`something-dir`) is repeated below.
This isn't very resilient to accidental bug introductions, these lists can get out of sync (e.g. spelling mistakes).

Please do it the Haskell way:  Define an enum type with the 4 options and have a single place where you converts these into strings and back.

Reply:
The *best* way would be to have a list of *functions* that retrieve the corresponding paths inside the configuration record, but Cabal's convoluted module structure made that impossible without circular imports.  That frustrated me so much that I just put in the strings, but I suppose I should go back and add that sum type.